### PR TITLE
[Mime] Fix memory leak

### LINKED ISSUE #51764: Memory Leak in Symfony Mailer when sending many attachments of >5MB
### Symfony version(s) affected

5.4.22

### Description

Hi , 
I have encountered an issue where attaching several files each of size 5MB , sends Out Of Memory exception 
` Uncaught PHP Exception Symfony\Component\ErrorHandler\Error\OutOfMemoryError: "Error: Allowed memory size of 33554432 bytes exhausted (tried to allocate 23079128 bytes)" at vendor/symfony/mime/RawMessage.php line 53 {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\OutOfMemoryError(code: 0): Error: Allowed memory size of 33554432 bytes exhausted (tried to allocate 23079128 bytes) at vendor/symfony/mime/RawMessage.php:53)"} [`

Is that a known issue. or am I missing something ? 

I have already tried using attachPart, attach , but it produces the same results.

### How to reproduce

Attach numerous Files from form to be sent in email each of size 5MB . 

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #51764
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/Mime/RawMessage.php [hunk 1]
```diff
@@ -48,12 +46,19 @@ public function toIterable(): iterable
             return;
         }
 
-        $message = '';
-        foreach ($this->message as $chunk) {
-            $message .= $chunk;
-            yield $chunk;
+        if ($this->message instanceof \Generator) {
```
Comment (fabpot) [inline src/Symfony/Component/Mime/RawMessage.php]: If the message is a generator, we can only use it once, so we need to store the result.

**Comment by fabpot** [comment]
Fixing it is not possible in 5.4.
I propose to deprecate the possibility of sending a message with a closed generator in 6.4 and throw in 7.0. That way, we don't need to store the generator result and that avoids the memory leak.

**Comment by stof** [comment]
@fabpot the issue is not just about sending a message with a closed generator. If you try to both access the string content and serialize the object, this will make 2 usages of the iterator, with the RawMessage being the one closing the generator in the first usage.

**Comment by fabpot** [comment]
@stof Sure, but that's not something people will do. What would you suggest?

**Comment by chalasr** [comment]
Isn't patching only `toIterable()` enough to fix the issue? 
```diff
diff --git a/src/Symfony/Component/Mime/RawMessage.php b/src/Symfony/Component/Mime/RawMessage.php
index d2a311daeb..14223d3d7f 100644
--- a/src/Symfony/Component/Mime/RawMessage.php
+++ b/src/Symfony/Component/Mime/RawMessage.php
@@ -48,12 +48,9 @@ class RawMessage implements \Serializable
             return;
         }
 
-        $message = '';
         foreach ($this->message as $chunk) {
-            $message .= $chunk;
             yield $chunk;
         }
-        $this->message = $message;
     }
 
     /**
```
Transforming the generator to an array in `toString()` seems unavoidable, unless we forbid calling toString() more than once which sounds worse.

**Pull request merged**